app/Services/RouteSuggester.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Venue;
+use Fisharebest\Algorithm\Dijkstra;
+use Illuminate\Database\Eloquent\Builder;
+
+class RouteSuggester
+{
+    private float $searchRange = 0.2;
+
+    public function __construct(
+        private array $stopsAttributes,
+        private array $start,
+        private array $end
+    ) {
+    }
+
+    public function suggest()
+    {
+        $stopsVenues =
+            $this->getLocalVenuesByAttributes(
+                $this->stopsAttributes,
+                $this->start,
+                $this->end,
+                $this->searchRange
+            );
+
+        $venuesGraph = $this->createVenuesGraph($stopsVenues);
+        $dijkstra = new Dijkstra($venuesGraph);
+        $path = $dijkstra->shortestPaths('start', 'end');
+
+        if (count($path) > 0) {
+            $ids = $path[0];
+            unset($ids[0]);
+            unset($ids[count($ids)]);
+
+            $venues = Venue::whereIn('id', $ids)->get();
+            return $venues;
+        } else return false;
```

Comment:
how are we handling if this fails?

Reply:
As in no path is found?

If there is no path return false, which then returns an empty array from the controller.

Reply:
Yes just some way of messaging or status if no path is found, that way we can know that there has been no path on the FE and return that information to the user in order for them to know why it has failed and what they could do next.